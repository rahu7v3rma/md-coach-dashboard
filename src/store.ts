import { configureStore } from '@reduxjs/toolkit';

import clientReducer from './reducers/client';
import groupReducer from './reducers/group';
import notificationReducer from './reducers/notification';
import userReducer from './reducers/user';

// define store options as a variable so we can export it for tests to use
const storeOptions: any = {
    reducer: {
        user: userReducer,
        client: clientReducer,
        notification: notificationReducer,
        group: groupReducer
    }
};

const store = configureStore(storeOptions);

export default store;

// infer the `RootState` type from the store
export type RootState = ReturnType<typeof store.getState>;

// infer the `AppDispatch` type from the store so thunk actions can return promises
export type AppDispatch = typeof store.dispatch;

export const test = {
    storeOptions
};
