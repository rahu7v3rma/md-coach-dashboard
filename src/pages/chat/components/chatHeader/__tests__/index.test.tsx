import React from 'react';
import { Provider } from 'react-redux';
import * as router from 'react-router';
import { BrowserRouter } from 'react-router-dom';
import { create } from 'react-test-renderer';

import ChatHeader from '../../chatHeader';
import store from 'src/store';

const mockOnCollapseButtonClick = jest.fn();
const navigate = jest.fn();

jest.mock('stream-chat-react', () => ({
    ...jest.requireActual('stream-chat-react'),
    useChannelStateContext: jest.fn(() => ({
        channel: {
            data: { name: 'Test' },
            state: {
                members: {}
            }
        }
    }))
}));

jest.mock('src/pages/chat/hooks', () => ({
    useChannelPreviewInfo: jest.fn(() => ({
        displayTitle: 'Test Channel'
    }))
}));

describe('ChatHeader', () => {
    beforeEach(() => {
        jest.spyOn(router, 'useNavigate').mockImplementation(() => navigate);
        jest.clearAllMocks();
        jest.spyOn(console, 'error').mockImplementation(() => {});
    });
    it('renders ChatHeader component with the correct title', () => {
        const tree = create(
            <Provider store={store}>
                <BrowserRouter>
                    <ChatHeader />
                </BrowserRouter>
            </Provider>
        );

        expect(tree.root).toBeTruthy();

        const chatTitle = tree.root.findByProps({
            'data-testid': 'chat-title'
        });
        expect(chatTitle.props.children).toBe('Test Channel');
    });

    it('calls onCollapseButtonClick function on collapse button click', () => {
        const component = create(
            <ChatHeader onCollapseButtonClick={mockOnCollapseButtonClick} />
        );

        component.root
            .findByProps({ 'data-testid': 'collapse-button' })
            .props.onClick();

        expect(mockOnCollapseButtonClick).toHaveBeenCalled();
    });

    it('renders collapse icon in the ChatHeader component', () => {
        const component = create(<ChatHeader />);
        const collapseButton = component.root.findByProps({
            'data-testid': 'collapse-button'
        });
        const collapseIcon = collapseButton.findByProps({
            'data-testid': 'collapse-icon'
        });

        expect(collapseIcon).toBeTruthy();
        expect(collapseIcon.props.alt).toBe('collapse');
    });

    it('does not throw an error when onCollapseButtonClick is not provided', () => {
        const errorSpy = jest.spyOn(console, 'error');
        errorSpy.mockImplementation(() => {});

        expect(errorSpy).not.toHaveBeenCalled();

        errorSpy.mockRestore();
    });
});
