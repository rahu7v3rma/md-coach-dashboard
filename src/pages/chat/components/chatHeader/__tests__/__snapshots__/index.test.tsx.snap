// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatHeader does not throw an error when onCollapseButtonClick is not provided 1`] = `
<div
  className="sc-aXZVg ghBzcq"
>
  <div
    className="sc-gEvEer fBFxJi"
    data-testid="chat-title"
  >
    Test Channel
  </div>
  <div
    className="sc-eqUAAy gPNtZy"
  >
    <button
      className="sc-fqkvVR bCyeDF"
      data-testid="collapse-button"
      onClick={[Function]}
    >
      <img
        alt="collapse"
        data-testid="collapse-icon"
        src={Object {}}
      />
    </button>
  </div>
</div>
`;

exports[`ChatHeader renders collapse icon in the ChatHeader component 1`] = `
<div
  className="sc-aXZVg ghBzcq"
>
  <div
    className="sc-gEvEer fBFxJi"
    data-testid="chat-title"
  >
    Test Channel
  </div>
  <div
    className="sc-eqUAAy gPNtZy"
  >
    <button
      className="sc-fqkvVR bCyeDF"
      data-testid="collapse-button"
      onClick={[Function]}
    >
      <img
        alt="collapse"
        data-testid="collapse-icon"
        src={Object {}}
      />
    </button>
  </div>
</div>
`;
